lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -216,6 +216,16 @@
     - ' - C(runonce) (list): List of commands to run at first user logon.'
     - ' - C(timezone) (int): Timezone (See U(https://msdn.microsoft.com/en-us/library/ms912391.aspx)).'
     version_added: '2.3'
+  vapp_properties:
+    description:
+    - A list of vApp properties
+    - 'For full list of attibutes and types refer to: U(https://github.com/vmware/pyvmomi/blob/master/docs/vim/vApp/PropertyInfo.rst)'
+    - 'Basic attributes are:'
+    - ' - C(id) (string): Property id - required'
+    - ' - C(value) (string): Property value'
+    - ' - C(type) (string): Value type, string type by default.'
+    - ' - C(operation) (string): add, edit, remove. Required only when removing properties. If property present the parameter will be defaulted to "edit".'
```

Comment:
From what I understand:
* "add" can only be used for adding a non existing property (it will fail on the second run)
* "edit" can be used to add a non existing property or edit an existing property
* "remove" can be used to remove an existing property and will not fail if the property doesn't exist.

Am I right ? If that so this should explicitly explain in the documentation.

Reply:
@tchernomax, thanks for pointing to this. I agree this is probably misleading or confusing. 
These are the three acceptable values for vim.option.ArrayUpdateSpec.Operation, but for the proposed code If operation is set to anything other than "remove" it's silently ignored.

Actual operation value calculated, thus no failures expected when user specify add for an existent property of trying to edit nonexistent one. For such cases, instead of failing we're just notifying user if changes were applied or not.

So, how about:  
    - ' - C(operation): C(remove). This attribute required only when removing properties.

Thanks.

Reply:
From your code:
* if the user set operation = add:
  * the prop. is present → FAIL, you don't redefine the operation before sending it to vim.option.ArrayUpdateSpec.Operation
  * the prop. isn't present → OK (it is created)
* if the user set operation = edit:
  * the prop. is present → OK (it is edited)
  * the prop. isn't present → OK (the operation re-defined as 'add')
* if the user set operation = remove:
  * the prop. is present → OK (it is removed)
  * the prop. isn't present → OK (nothing is done)

So there is still one FAIL.

Also I am OK with your proposition to change the operation:
> - ' - C(operation): C(remove). This attribute required only when removing properties.'

Reply:
You definitely right here, thanks. Made changes to both: the failure case, and the documentation.